packages/use-sync-external-store/src/__tests__/useSyncExternalStoreShared-test.js

```diff
@@ -594,6 +598,8 @@ describe('Shared useSyncExternalStore behavior (shim and built-in)', () => {
       );
     }).toErrorDev(
       'The result of getSnapshot should be cached to avoid an infinite loop',
+      // Stacks don't work when mixing the source and the npm package.
+      {withoutStack: gate(flags => flags.enableUseSyncExternalStoreShim)},
```

Comment:
Seems like they do work in more scenarios: 

```
FAIL packages/use-sync-external-store/src/__tests__/useSyncExternalStoreShared-test.js
  ● Shared useSyncExternalStore behavior (shim and built-in) › Infinite loop if getSnapshot keeps returning new reference

    Received warning unexpectedly includes a component stack:
      "Warning: The result of getSnapshot should be cached to avoid an infinite loop
        in App (at **)"
    If this warning intentionally includes the component stack, remove {withoutStack: true} from the toErrorDev() call. If you have a mix of warnings with and without stack in one toErrorDev() call, pass {withoutStack: N} where N is the number of warnings without stacks.

      597 |           'the number of nested updates to prevent infinite loops.',
      598 |       );
    > 599 |     }).toErrorDev(
          |        ^
      600 |       'The result of getSnapshot should be cached to avoid an infinite loop',
      601 |       // Stacks don't work when mixing the source and the npm package.
      602 |       {withoutStack: gate(flags => flags.enableUseSyncExternalStoreShim)},

      at Object.<anonymous> (packages/use-sync-external-store/src/__tests__/useSyncExternalStoreShared-test.js:599:8)
```
https://app.circleci.com/pipelines/github/facebook/react/49531/workflows/85823153-d4e2-4c91-96e3-4df0ba4ab35b/jobs/755631?invite=true#step-106-1266_121

Reply:
Yeah, it works in the builds, and I also need to use the prod UMD builds in the prod tests.